runtimes/pythonrt/requirements.go

```diff
@@ -0,0 +1,37 @@
+package pythonrt
+
+import (
+	"archive/tar"
+	"bytes"
+	"errors"
+	"io"
+
+	"go.autokitteh.dev/autokitteh/internal/kittehs"
+)
+
+func getRequirementsHash(artifact []byte) (string, error) {
+	tf := tar.NewReader(bytes.NewReader(artifact))
+
+	for {
+		hdr, err := tf.Next()
+		if errors.Is(err, io.EOF) {
+			return "", nil
+		}
+
+		if err != nil {
+			return "", err
+		}
+
+		if hdr.Name != "requirements.txt" {
+			continue
+		}
+
+		var b bytes.Buffer
+
+		if _, err := io.Copy(&b, tf); err != nil {
+			return "", err
+		}
+
+		return kittehs.SHA256Hash(b.Bytes())
```

Comment:
`kittehs.SHA256Hash` gobs encode the value first. IMO this can be not deterministic.
From the [gob documentation](https://go.dev/blog/gob#types-on-the-wire):

> The first time you send a given type, the gob package includes in the data stream a description of that type. In fact, what happens is that the encoder is used to encode, in the standard gob encoding format, an internal struct that describes the type and gives it a unique number.

Use sha256 (or just sha1) on the bytes.

Reply:
fixed